### PR TITLE
Feat: Support all proposal types from archive data

### PR DESCRIPTION
**Pull request by sudheerDev**
feat(proposals): implement type-safe archive proposal system with unified normalization and status derivation

Refactor the proposal data layer to support all archive-backed proposal types by introducing a single, type-safe normalization pipeline from DAO Node, EAS Atlas/OODAO, and Snapshot sources into the shared Proposal model used across list and detail views. Add source-aware extractors, reusable conversion/formatting helpers, and voting-type-specific status derivation (standard, approval, optimistic, tiered optimistic) with explicit threshold/quorum logic and broad test coverage for edge cases.

Replace fragmented archive normalization paths and ad-hoc UI branching with a declarative proposal registry and specialized row/detail components, improving consistency of status rendering and vote summaries across proposal pages. Remove deprecated normalization utilities, align tenant proposal type mappings/configuration for the new model, and document proposal type behavior and field mappings to make future proposal-type additions safer and easier.

### CODE REVIEW

File: src/app/api/proposals/[proposalId]/copeland-result/route.ts [hunk 1]
```diff
@@ -60,18 +95,27 @@ export async function GET(
...
-      fetchSnapshotVotesForProposal({
-        proposalId: route.params.proposalId,
-        pagination: {
-          offset: 0,
-          limit: 100000,
-        },
-      }),
+    const { namespace } = Tenant.current();
+    const proposalId = route.params.proposalId;
+
```
[the remark this answers is not in the document]
Comment (sudheerDev) [reply src/app/api/proposals/[proposalId]/copeland-result/route.ts:100]: it is not sensitive data

File: src/app/api/proposals/[proposalId]/copeland-result/route.ts [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
+  fetchRawProposalVotesFromArchive,
+} from "@/lib/archiveUtils";
+import { archiveToProposal } from "@/lib/proposals";
+import { Proposal } from "@/app/api/common/proposals/proposal";
+import { SnapshotVote } from "@/app/api/common/votes/vote";
+import { fetchProposalTaxFormMetadata } from "@/app/api/common/proposals/getProposalTaxFormMetadata";
+
+async function loadProposal(proposalId: string): Promise<Proposal> {
+  const { namespace, token, ui } = Tenant.current();
+  const useArchive = ui.toggle("use-archive-for-proposal-details")?.enabled;
```
[the remark this answers is not in the document]
Comment (sudheerDev) [reply src/app/api/proposals/[proposalId]/copeland-result/route.ts:17]: Not sensitive
